src/SuperVault.sol

```diff
@@ -594,11 +682,56 @@ contract SuperVault is BaseStrategy, ISuperVault {
         /// @notice assign remaining weight to the last index
         newWeights[length - 1] = TOTAL_WEIGHT - totalAssignedWeight;
 
-        /// @dev update SV weights
-        SV.weights = newWeights;
-        SV.superformIds = finalSuperformIds;
-        SV.numberOfSuperforms = length;
+        /// @dev update SV data
+        weights = newWeights;
+        superformIds = finalSuperformIds;
+        numberOfSuperforms = length;
 
         emit RebalanceComplete(finalSuperformIds, newWeights);
     }
+
+    /// @notice Changes the whitelist for a Superform ID
+    /// @param superformId The Superform ID to change
+    /// @param isWhitelisted Whether to whitelist or blacklist
+    function _changeSuperformWhitelist(uint256 superformId, bool isWhitelisted) internal {
+        bool currentlyWhitelisted = whitelistedSuperformIds[superformId];
+
+        // Only process if there's an actual change
+        if (currentlyWhitelisted != isWhitelisted) {
+            whitelistedSuperformIds[superformId] = isWhitelisted;
+
+            if (isWhitelisted) {
+                _addToWhitelist(superformId);
+            } else {
+                _removeFromWhitelist(superformId);
+            }
+
+            emit SuperformWhitelisted(superformId, isWhitelisted);
+        }
+    }
+
+    /// @notice Adds a superform ID to the whitelist array
+    /// @param superformId The Superform ID to add
+    function _addToWhitelist(uint256 superformId) internal {
+        whitelistedSuperformIds[superformId] = true;
```

Comment:
could check the `superformId` isn't already present to prevent adding duplicates
(in that case `whitelistedSuperformIdArray[]` will have the same entry twices which might give some weird results later on)

Reply:
https://github.com/superform-xyz/SuperVaults/pull/31